pkg/clusteragent/admission/mutate/autoinstrumentation/target_mutator.go

```diff
@@ -178,24 +181,28 @@ func (m *TargetMutator) MutatePod(pod *corev1.Pod, ns string, _ dynamic.Interfac
 	}
 	extracted := m.core.initExtractedLibInfo(pod).withLibs(target.libVersions)
 
-	// Add the configuration for the security client library.
-	for _, mutator := range m.securityClientLibraryPodMutators {
-		if err := mutator.mutatePod(pod); err != nil {
-			return false, fmt.Errorf("error mutating pod for security client: %w", err)
-		}
+	if err := allContainersPodMutator(
+		m.containerFilter,
+		m.securityClientLibraryMutators,
+	).mutatePod(pod); err != nil {
+		return false, fmt.Errorf("error mutating pod for security client: %w", err)
 	}
 
-	// Add the configuration for profiling.
-	for _, mutator := range m.profilingClientLibraryPodMutators {
-		if err := mutator.mutatePod(pod); err != nil {
-			return false, fmt.Errorf("error mutating pod for profiling client: %w", err)
-		}
+	if err := allContainersPodMutator(
+		m.containerFilter,
+		m.profilingClientLibraryMutators,
+	).mutatePod(pod); err != nil {
+		return false, fmt.Errorf("error mutating pod for profiling client: %w", err)
 	}
 
-	// Inject the tracer configs. We do this before lib injection to ensure DD_SERVICE is set if the user configures it
-	// in the target.
+	// Inject the tracer configs.
+	// We do this before lib injection to ensure DD_SERVICE is set
+	// if the user configures it in the target.
 	for _, envVar := range target.envVars {
-		mutatecommon.InjectEnv(pod, envVar)
+		_ = allContainersPodMutator(
```

Comment:
If I'm being honest, I really find that these mutators hard to reason about. I find it much more straightforward when we have a methods like `InjectEnv` that add the env var to a pod. Reading the `allContainersPodMutator`, it's really not clear what is happening. 

Would it be more clear to simply pass a map into `mutatecommon.InjectEnv(pod, envVar, containerFilter)`?

Reply:
Agree this name is bad and the mutators are a bit much here.

I can try that-- the one other thing that occurred to me about this function was that it doesn't need to exist :) alternatively a `mutatePodContainers(pod, containerMutator)` would be clearer